src/coreclr/jit/valuenum.cpp

```diff
@@ -1634,6 +1634,13 @@ bool ValueNumStore::IsKnownNonNull(ValueNum vn)
     {
         return false;
     }
+
+    if (IsVNObjHandle(vn))
```

Comment:
Why just `ObjHandle`?

I think it could be any handle, as in `fgAddrCouldBeNull`.

Reply:
Good point! since none of them are allowed to be zero.